src/io/metadata.cpp

```diff
@@ -528,6 +595,34 @@ void Metadata::LoadWeights() {
   weight_load_from_file_ = true;
 }
 
+void Metadata::LoadPositions() {
+  num_positions_ = 0;
+  if (position_filename_ == std::string("")) {
```

Comment:
Should this be moved up *before* setting `num_positions_ = 0`? That'd avoid a tiny bit of unnecessary computation, and would prevent bugs of the form "`num_positions_` was set to 0 even though `positions` was provided through an in-memory array" less likely.

Reply:
Done